ga4gh/cli.py

```diff
@@ -16,6 +16,109 @@
+        raise NotImplementedError()
+
+    @classmethod
+    def createGASearchVariantSetsRequest(cls, host, args):
+        request = protocol.GASearchVariantSetsRequest()
+        setCommaSeparatedAttribute(request, args, 'datasetIds')
+        request.pageSize = args.pageSize
+        request.pageToken = None
+        return request
+
+    @classmethod
+    def createGASearchVariantsRequest(cls, host, args):
+        request = protocol.GASearchVariantsRequest()
+        request.referenceName = args.referenceName
+        request.variantName = args.variantName
+        request.start = args.start
+        request.end = args.end
+        if host.usingWorkaroundsFor(client.HttpClient.workaroundGoogle):
+            request.maxCalls = args.maxCalls
+        if args.callSetIds == []:
+            request.callSetIds = []
+        elif args.callSetIds == '*':
+            request.callSetIds = None
+        else:
+            request.callSetIds = args.callSetIds.split(",")
+        setCommaSeparatedAttribute(request, args, 'variantSetIds')
+        return request
+
+    @classmethod
+    def createGASearchReferenceSetsRequest(cls, host, args):
+        request = protocol.GASearchReferenceSetsRequest()
+        setCommaSeparatedAttribute(request, args, 'accessions')
+        setCommaSeparatedAttribute(request, args, 'md5checksums')
+        return request
+
+    @classmethod
+    def createGASearchReferencesRequest(cls, host, args):
+        request = protocol.GASearchReferencesRequest()
+        setCommaSeparatedAttribute(request, args, 'accessions')
+        setCommaSeparatedAttribute(request, args, 'md5checksums')
+        return request
+
+    @classmethod
+    def createGASearchReadGroupSetsRequest(cls, host, args):
+        request = protocol.GASearchReadGroupSetsRequest()
+        setCommaSeparatedAttribute(request, args, 'datasetIds')
+        request.name = args.name
+        return request
+
+    @classmethod
+    def createGASearchCallSetsRequest(cls, host, args):
+        request = protocol.GASearchCallSetsRequest()
+        setCommaSeparatedAttribute(request, args, 'variantSetIds')
+        request.name = args.name
+        return request
+
+    @classmethod
+    def createGASearchReadsRequest(cls, host, args):
+        request = protocol.GASearchReadsRequest()
+        if host.usingWorkaroundsFor(client.HttpClient.workaroundGoogle):
+            # google says referenceId not a valid field
+            request = cls.GASearchReadsRequestGoogle()
+        setCommaSeparatedAttribute(request, args, 'readGroupIds')
+        request.start = args.start
+        request.end = args.end
+        request.referenceId = args.referenceId
+        request.referenceName = args.referenceName
+        return request
+
+    @classmethod
+    def createGAListReferenceBasesRequest(cls, host, args):
+        request = protocol.GAListReferenceBasesRequest()
+        request.start = args.start
+        request.end = args.end
+        host._id = args.id
```

Comment:
This is a violation of the isolation rules. If we need to set this `id` value, then this should be a method of `AbstractSearchRunner`, so that we can legitimately access `self._id`.